test/new-e2e/agent-subcommands/secret_test.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package agentsubcommands
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+)
+
+type agentSecretSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestAgentSecretSuite(t *testing.T) {
+	e2e.Run(t, &agentSecretSuite{}, e2e.AgentStackDef(nil))
+}
+
+func (v *agentSecretSuite) TestAgentSecretNotEnabledByDefault() {
+	v.UpdateEnv(e2e.AgentStackDef(nil))
+
+	secret := v.Env().Agent.Secret()
+
+	assert.Contains(v.T(), secret, "No secret_backend_command set")
+}
+
+func (v *agentSecretSuite) TestAgentSecretChecksExecutablePermissions() {
+	v.UpdateEnv(e2e.AgentStackDef(nil, agentparams.WithAgentConfig("secret_backend_command: /usr/bin/echo")))
+
+	output := v.Env().Agent.Secret()
+
+	assert.Contains(v.T(), output, "=== Checking executable permissions ===")
```

Comment:
Is there a particular reason to not just compare the full output to an expected string ?
With the current implementation new lines would be ignored, and the output format (ie. the order of the strings, spacing, empty lines) is not checked either

Reply:
my thinking was to not tie the test to the template changes, we can follow up in a new PR if we want to change the approach